apps/prairielearn/src/pages/instructorAssessmentInstances/instructorAssessmentInstances.html.ts

```diff
@@ -0,0 +1,583 @@
+  score_perc: AssessmentInstanceSchema.shape.score_perc.nullable(),
+  time_remaining_sec: z.number().nullable(),
+  time_remaining: z.string(),
+  total_time_sec: z.number().nullable(),
+  total_time: z.string(),
+  uid_list: z.array(UserSchema.shape.uid).nullable(),
+  uid: UserSchema.shape.uid.nullable(),
+  user_id: UserSchema.shape.user_id.nullable(),
+  user_name_list: z.array(UserSchema.shape.name).nullable(),
+  username: z.string().nullable(),
+});
+export type AssessmentInstanceRow = z.infer<typeof AssessmentInstanceRowSchema>;
+
+export function InstructorAssessmentInstances({ resLocals }: { resLocals: Record<string, any> }) {
+  return html`
+    <!doctype html>
+    <html lang="en">
+      <head>
+        ${renderEjs(__filename, "<%- include('../partials/head'); %>", resLocals)}
+
+        <script src="${nodeModulesAssetPath(
+            'bootstrap-table/dist/bootstrap-table.min.js',
+          )}"></script>
+        <script src="${nodeModulesAssetPath(
+            'bootstrap-table/dist/extensions/sticky-header/bootstrap-table-sticky-header.min.js',
+          )}"></script>
+        <script src="${nodeModulesAssetPath(
+            'bootstrap-table/dist/extensions/auto-refresh/bootstrap-table-auto-refresh.js',
+          )}"></script>
+        <link
+          href="${nodeModulesAssetPath('bootstrap-table/dist/bootstrap-table.min.css')}"
+          rel="stylesheet"
+        />
+        <link
+          href="${nodeModulesAssetPath(
+            'bootstrap-table/dist/extensions/sticky-header/bootstrap-table-sticky-header.min.css',
+          )}"
+          rel="stylesheet"
+        />
+
+        ${compiledScriptTag('instructorAssessmentInstancesClient.ts')}
+      </head>
+      <style>
+        .sticky-column {
+          position: sticky;
+          left: 0;
+          background: white;
+          background-clip: padding-box;
+          box-shadow: inset -1px 0 #dee2e6;
+        }
+        .table-hover tbody tr:hover td.sticky-column {
+          color: #212529;
+          background-color: #efefef;
+        }
+        .fixed-table-toolbar {
+          padding: 0 1em 0 1em;
+        }
+        .fixed-table-toolbar div.pagination,
+        .fixed-table-toolbar div.pagination-detail {
+          margin: 0 1em 0 0 !important;
+        }
+      </style>
+      <body>
+        ${renderEjs(__filename, "<%- include('../partials/navbar'); %>", resLocals)}
+        <main id="content" class="container-fluid">
+          ${renderEjs(
+            __filename,
+            "<%- include('../partials/assessmentSyncErrorsAndWarnings'); %>",
+            resLocals,
+          )}
+          ${resLocals.authz_data.has_course_instance_permission_edit
+            ? html`
+                <div
+                  class="modal fade"
+                  id="deleteAssessmentInstanceModal"
+                  tabindex="-1"
+                  role="dialog"
+                  aria-labelledby="deleteAssessmentInstanceModalLabel"
+                >
```

Comment:
Use `Modal(...)` here? Ditto for others on the page. There are enough modals on this page that I'd move them to other functions, e.g.:

```ts
function DeleteAssessmentInstanceModal() {
  return html`...`;
}
```

Reply:
Added `Modal` component here and for the other modals on the page in [306ccbe](https://github.com/PrairieLearn/PrairieLearn/pull/9725/commits/306ccbe9c0ebd4481747d7e7ddce945db6fff328).

Reply:
I believe not all modals were converted in the commit above...

Reply:
Remaining modals were converted in [7efdf17](https://github.com/PrairieLearn/PrairieLearn/pull/9725/commits/7efdf1794fea6c432e21d53c25ae86cac0585ebd).